### PR TITLE
[WEB-3364] Storybook migration: Add Storybook

### PR DESCRIPTION
**Pull request by dpiatek**
**Update (18/3/24), jamiehenson:**
This PR fundamantally does the following things:
- Adds Storybook as a dependency and sets up preliminary configs and theming
- Adds `[COMPONENT_NAME].stories.tsx` files that contain Stories for each of the components represented in the Rails `ably-ui`
- To support this, TypeScript linting support has also been set up here
- In reference to the quandaries below, I chose source components over compiled. As a caveat to that, I've removed usages of Tailwind's `@layer` directive as it was a blocker here (despite what various Storybook/Tailwind/PostCSS blogs and docs promise), and ultimately not something (in my view) that would necessitate the massive workarounds needed to make it play nice with SB.

Cypress is inactive with this release, the tests need to be rewritten - probably as part of #323.

To test, pull down, grab the latest packages, and run `yarn storybook` from root.

**Original PR description:**
Replace the preview server with Storybook.

Explore what kind of issues this kind of migration would encounter:
- tailwind continues to be a bit of pain whenever any kind of build setup is involved; definitely need to simplify all of that setup both here and for consumers
- a choice will need to be made whether to use compiled components or use the ones from source:
   - source has prop-types which allow us to autogenerate docs to some extent
   - hot reload is likely easier with source
   - CSS reload might be harder
   - compiled components will require a side process to run webpack and compile things
- mock services will have to be written for things like sessions and blog posts
- not completely sure we can move off cypress to storybook - would have to research that more

### CODE REVIEW

File: tailwind.config.js [hunk 1]
```diff
@@ -1,17 +1,11 @@
...
-    "./src/**/*.css",
-  ],
+  content: ["./src/**/*.{jsx,js,json,css,ts,tsx,mdx}"],
   safelist: [
     "w-1/2",
     "w-1/3",
     "w-1/4",
     "w-1/5",
     "w-1/6",
-    { pattern: /^hljs.*/ },
```
Comment (kennethkalmer) [inline tailwind.config.js:14]: These are the syntax highlighter classes, is this okay to remove?

**Review by kennethkalmer — CHANGES_REQUESTED** [review]
This looks great @jamiehenson (and @dpiatek), thanks so much!

A few small comments to give it some shine :)

* Squash the existing featured link fixup commit down
* Squash `chore: don't allow SB version ads or telemetry` into `Add storybook scaffold`
* Drop the first commit, then `chore: resolve preview merge commit and remove TW base layer usage` will be reduced to only the meganav test data comment, and can be squashed into `feat: port existing ably-ui content to SB stories, strip out TW layer…`

(Be careful with the squashes, I can help if that would make it more comfortable)

**Comment by jamiehenson** [comment]
@kennethkalmer done the squashing, thanks for the suggestions.

**Review by kennethkalmer — APPROVED** [review]

**Pull request merged**